backend/src/helpers/databasefunctions.ts

```diff
@@ -16,31 +17,156 @@ const pool = new Pool(pgconfig);
 
 pool.on(
   'error',
-  async (_err: Error): Promise<never> => {
-    throw new ErrorHandler(400, { status: 'Database Error' });
+  async (err: Error): Promise<void> => {
+    // Uncertain how to properly solve this problem.
+    await parsePgError((err as unknown) as PgError);
   },
 );
 
+const hasConnection = async () => {
+  try {
+    await (await pool.connect()).query('SELECT NOW()');
+  } catch (e) {
+    await parsePgError(e);
+  }
+};
+
+const getClient = async (connection: Pool | Client): Promise<PoolClient | Client | undefined> => {
+  let client: any;
+  if (isPool(connection)) {
+    client = await connection.connect();
+    return client as PoolClient;
+  } else if (isClient(connection)) {
+    client = connection;
+    return client as Client;
+  } else {
+    client = undefined;
+    return client as undefined;
+  }
+};
+
+const disconnectClient = async (client: PoolClient | Client | undefined) => {
+  if (client === undefined) {
+    //TODO
```

Comment:
It should be impossible for the code to reach that point but because of TS compile errors i had to have that if block. I'm uncertain if i need to add some code in there... The TODO was to reread the code and check.

Reply:
I think the solution is to remove the error. Seems like it should be impossible for the code to reach that if block, but i need it to compile with strict settings.